src/py/crankshaft/crankshaft/analysis_data_provider.py

```diff
@@ -2,18 +2,25 @@
 import plpy
 import pysal_utils as pu
 
+NULL_VALUE_ERROR = ('No usable data passed to analysis. Check your input rows '
+                    'for null values and fill in appropriately.')
+
+
+def verify_data(n_rows):
```

Comment:
👍 agree. Since there is only one check now I was thinking it'd save a bit of time on passing around all the data that could be in `data`, but that timing is probably pretty minimal compared to other things in the code.